compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -370,7 +370,7 @@ impl<Bytes: AllocBytes> Allocation<CtfeProvenance, (), Bytes> {
         }
         // Create allocation.
         Ok(Allocation {
-            bytes,
+            bytes: AllocBytes::from_bytes(Cow::Owned(Vec::from(bytes)), self.align),
```

Comment:
```suggestion
            bytes: AllocBytes::from_bytes(Cow::Owned(bytes), self.align),
```

Let the `<Bytes as AllocBytes>::from_bytes` handle creating the heap allocation

Reply:
We start with a heap allocation (`bytes` is a `Box`) and we want to enable `from_bytes` to reuse that.